### PR TITLE
feat(Backend + SDK): Update kfp backend and kubernetes sdk to support ImagePullPolicy

### PR DESCRIPTION
**Pull request by revit13**
Description of your changes:
Part of Part of https://github.com/kubeflow/pipelines/issues/9714 and https://github.com/kubeflow/pipelines/issues/9980

Update kfp backend and kubernetes sdk to support image pull policy.
To be rebased and update go.mod after #10416 

Checklist:

 The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by Tomcli** [comment]
rebase and update go.mod after #10416

**Review by Tomcli — CHANGES_REQUESTED** [review]
@revit13 Can you update the go.mod and the related files like in this PR? Thanks.
https://github.com/kubeflow/pipelines/pull/10427#pullrequestreview-1868388426

File: backend/src/v2/driver/driver.go [hunk 1]
```diff
@@ -475,6 +475,12 @@ func extendPodSpecPatch(
 		podSpec.Containers[0].VolumeMounts = append(podSpec.Containers[0].VolumeMounts, volumeMounts...)
 	}
 
+	// Get image pull policy
+	if kubernetesExecutorConfig.GetImagePullPolicy() != "" {
```
Comment (Tomcli) [inline backend/src/v2/driver/driver.go:479]: Maybe we can also set kubernetesExecutorConfig.GetImagePullPolicy() as a variable, then check whether the value is one of ['Always', 'Never', 'IfNotPresent']?

There could be a case where someone can compose the IR without the SDK, so we also wants to check the value is valid before passing into the pod spec. If the value is not valid, we should raise an error instead.

Comment (revit13) [reply backend/src/v2/driver/driver.go]: @Tomcli Done. Thanks

**Comment by chensun** [comment]
@revit13 Can you please rebase and resolve the conflicts?

File: kubernetes_platform/python/kfp/kubernetes/image.py [hunk 2]
```diff
@@ -38,11 +38,33 @@ def set_image_pull_secrets(
...
 
     return task
+
+
+def set_image_pull_policy(task: PipelineTask, policy: str) -> PipelineTask:
+    """Set image pull policy for the container.
+
+    Args:
+        task: Pipeline task.
+        policy: One of `Always`, `Never`, `IfNotPresent`.
```
Comment (chensun) [inline kubernetes_platform/python/kfp/kubernetes/image.py:57]: nit: using enum could be less error-prone, and also works with IDE code auto-complete. No strong preference though.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

**Comment by Hammad-NobleAI** [comment]
Putting this comment here... we need this facility through the Python API. Do we have any idea when the next release might be?

**Pull request merged**